pkgconfig flag is not usable

Currently if the pkgconfig flag is enabled, Cabal fails to parse the somewhat strange libcmark-gfm version:

```
Setup: parsing output of pkg-config --modversion failed
```

The output of that command is:

```
$ pkg-config --modversion libcmark-gfm
0.28.3.gfm.14
```

However, if I change that to a simple link with `Extra-Libraries: cmark-gfm cmark-gfmextensions`, the test suite fails to link:

```
Building test suite 'test-cmark-gfm' for cmark-gfm-0.1.4..
[1 of 1] Compiling Main             ( test/test-cmark.hs, dist/build/test-cmark-gfm/test-cmark-gfm-tmp/Main.dyn_o )
Linking dist/build/test-cmark-gfm/test-cmark-gfm ...
/build/haskell-cmark-gfm/src/cmark-gfm-0.1.4/dist/build/libHScmark-gfm-0.1.4-LX6NmMSQV2654l7JqEAgyX-ghc8.4.3.so: error: undefined reference to 'CMARK_DEFAULT_MEM_ALLOCATOR'
collect2: error: ld returned 1 exit status
`gcc' failed in phase `Linker'. (Exit code: 1)
```

It seems the symbol is not present in the .so file, and can only be found in the source .c files.

Is it possible to make cmark-gfm-hs buildable with system libcmark-gfm?

--- a/CMarkGFM.hsc
+++ b/CMarkGFM.hsc
@@ -58,7 +58,7 @@ ensurePluginsRegistered = c_core_extensions_ensure_registered
 
 -- | Frees a cmark linked list, produced by extsToLlist.
 freeLlist :: LlistPtr a -> IO ()
-freeLlist = c_cmark_llist_free c_CMARK_DEFAULT_MEM_ALLOCATOR
+freeLlist = c_cmark_llist_free c_cmark_mem
 
 -- | Converts a list of resolved extension pointers to a single cmark
 -- linked list, which can be passed to functions requiring a list of
@@ -67,7 +67,7 @@ extsToLlist :: [ExtensionPtr] -> IO (LlistPtr ExtensionPtr)
 extsToLlist [] = return nullPtr
 extsToLlist (h:t) = do
   t' <- extsToLlist t
-  c_cmark_llist_append c_CMARK_DEFAULT_MEM_ALLOCATOR t' (castPtr h)
+  c_cmark_llist_append c_cmark_mem t' (castPtr h)
 
 -- | Resolves CMarkExtensions to pointers.
 resolveExts :: [CMarkExtension] -> IO [ExtensionPtr]
@@ -633,8 +633,8 @@ foreign import ccall "cmark.h cmark_llist_append"
 foreign import ccall "cmark.h cmark_llist_free"
     c_cmark_llist_free :: MemPtr -> LlistPtr a -> IO ()
 
-foreign import ccall "cmark.h &CMARK_DEFAULT_MEM_ALLOCATOR"
-    c_CMARK_DEFAULT_MEM_ALLOCATOR :: MemPtr
+foreign import ccall "cmark.h cmark_get_default_mem_allocator"
+    c_cmark_mem :: MemPtr
 
 foreign import ccall "cmark_extension_api.h cmark_parser_attach_syntax_extension"
     c_cmark_parser_attach_syntax_extension :: ParserPtr -> ExtensionPtr -> IO ()

--- a/cmark-gfm.cabal
+++ b/cmark-gfm.cabal
@@ -74,7 +74,7 @@ library
   default-language:    Haskell2010
   ghc-options:         -Wall -fno-warn-unused-do-bind
   if flag(pkgconfig)
-    pkgconfig-depends: libcmark-gfm
+    Extra-Libraries: cmark-gfm cmark-gfmextensions
   else
     cc-options:        -Wall -std=c99
     Include-dirs:      cbits

